### DISCUSSION TITLE
Help with passing ID to WireUI modal

### DISCUSSION THREAD

**Discussion by gldtn** [rappasoft/laravel-livewire-tables#722] [Q&A]
I´m trying to create a user CRUD on the UsersTable and I'm not sure if I'm going the right direction here..

My delete function is as follow..

**App\Http\Livewire\UserCrud.php** 

```
public function destroy($id)
    {
        User::find($id)->delete();
        //session()->flash('message', 'User deleted.');
    }
```

**App\Http\Livewire\UsersTable,php** 

```
public function delete($id) {
        (new UserCrud())->destroy($id);
    }
```

**And..**

```
Column::make('Actions')
                ->label(function($row) {
                    return view('livewire.user.actions')->withRow($row);
                })
```

**resources\views\livewire\user\actions.blade.php**

```
<div class="flex justify-middle">
    
    <div>
        <x-button.circle xs negative icon="x" onclick="$openModal('deleteUserModal')" />
    </div>
</div>
```

**And..**

```
<x-modal.card title="Edit User" blur wire:model.defer="deleteUserModal">

    <div class="grid grid-cols-1 sm:grid-cols-2 gap-4">

        Are you sure you would like to DELETE user

    </div>

    <x-slot name="footer">

        <div class="flex justify-end gap-x-4">

            <div class="flex">

                <x-button flat label="Cancel" x-on:click="close" />

                <x-button negative label="Delete" wire:click="delete({{ $id }})" />

            </div>

        </div>

    </x-slot>

</x-modal.card>
```

**And I get**

`Undefined variable $id`

**I also tried:**
`wire:click="delete({{ $row->id }})"`

**But I get:**
`Unable to resolve dependency [Parameter #0 [ <required> $id ]] in class App\Http\Livewire\UsersTable`

How do I pass ID to my modal? Can anyone shed a light please?

**Comment by dnielandt** [comment]
Had a simular issue. 
You try override this by 

[https://rappasoft.com/docs/laravel-livewire-tables/v2/datatable/available-methods#content-builder](url)

`$this->setAdditionalSelects(['users.id as id']);`

Hope this helps...
